core/src/main/scala/akka/persistence/cassandra/journal/CassandraJournal.scala

```diff
@@ -286,18 +286,25 @@ class CassandraJournal(cfg: Config)
         result.map(_ => extractTagWrites(serialized))
     }
 
-    bulkTagWrite.onComplete { _ =>
+    val toReturn = bulkTagWrite.map(btw => {
+      // notify TagWriters when write was successful before completing the future otherwise
+      // we can get another seq of AtomicWrites for the same persistent actor before this is sent
+      tagWrites.foreach(_ ! btw)
       self ! WriteFinished(pid, writeInProgressForPersistentId.future)
       writeInProgressForPersistentId.success(Done)
-    }(akka.dispatch.ExecutionContexts.sameThreadExecutionContext)
-
-    //Nil == all good
-    bulkTagWrite.map(bulkTagWrite => {
-      // notify TagWriters when write was successful before completing the future meaning
-      // that we can get another seq of AtomicWrites for the same persistent actor
-      tagWrites.foreach(_ ! bulkTagWrite)
+      //Nil == all good
       Nil
     })(akka.dispatch.ExecutionContexts.sameThreadExecutionContext)
+
+    // if the write fails still need to remove state from the map
+    toReturn.onComplete {
```

Comment:
`.failed.foreach`, and it would be nice to DRY the two lines into a `def`, but my nitpicking quota is full for this PR

Reply:
Yep that was sloppy, fixed